app/components/web-push-subscription/component.js

```diff
@@ -0,0 +1,71 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import urlBase64ToUint8Array from 'hyperchannel/utils/url-base64-to-uint8-array';
+
+export default Component.extend({
+
+  botkaURL: null,
+  username: null,
+  subscribed: false,
+
+  webPushBaseURL: computed('botkaURL', function() {
+    return this.botkaURL + '/web-push';
+  }),
+
+  init () {
+    this._super(...arguments);
+
+    navigator.serviceWorker.ready.then(registration => {
+      return registration.pushManager.getSubscription();
+    }).then(subscription => {
+      if (subscription) { this.set('subscribed', true); }
+    });
+  },
+
+  actions: {
+
+    subscribe () {
+      navigator.serviceWorker.ready.then(async function(registration) {
+        // Get the server's public key
+        const response = await fetch(this.webPushBaseURL+'/vapid-public-key');
+        const vapidPublicKey = await response.text();
+        // Chrome doesn't accept the base64-encoded (string) vapidPublicKey yet
+        // urlBase64ToUint8Array() is defined in /tools.js
+        const convertedVapidKey = urlBase64ToUint8Array(vapidPublicKey);
+        // Subscribe the user
+        return registration.pushManager.subscribe({
+          userVisibleOnly: true,
+          applicationServerKey: convertedVapidKey
+        });
+      }.bind(this)).then(subscription => {
```

Comment:
Do fat arrow functions not work with `async`?

Reply:
Oh, this might be a leftover from the code I copied. Might work with a fat arrow, too.